### PR TITLE
Add coverage threshold for simplecov.

### PR DESCRIPTION
**Pull request by razanjoshi**
#### What? Why?

This PR makes sure there is a minimum level of tests covered by the developers. I have set the threshold at 54 for now but can be re-adjusted manually using SimpleCov coverage helper.

I initially set the threshold to 94% but since the lack of specs, the CI build fails. Hence reduced to 54% now as that is the current test coverage of the project. Anyone can add more specs to increase the test coverage.
Screenshot 2020-10-11 at 19 17 04

#### What should we test?
This creates test coverage report for test and dev env, which gives more idea of the coverage of specs needed or the specs that have covered various part of the codebase.

### CODE REVIEW

File: spec/coverage_helper.rb [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+
+SimpleCov.minimum_coverage 54
+SimpleCov.start 'rails' do
+  add_filter '/bin/'
+  add_filter '/config/'
+  add_filter '/jobs/application_job.rb'
+  add_filter '/schemas/'
+  add_filter '/lib/generators'
+  add_filter '/spec/'
+  add_filter '/vendor/'
```
Comment (luisramos0) [inline spec/coverage_helper.rb:13]: I wonder if we need to add more entries here, like:
/public
/swagger
/script
/log
/db

**Review by luisramos0 — COMMENTED** [review]
Nice, I think this is a good thing to have.

I just wonder what are those 17klocs...

I see at least ~40klocs:
~/dev/ofn (master) $ find app -name *.rb | xargs cat | wc -l
24490
~/dev/ofn (master) $ find lib -name *.rb | xargs cat | wc -l
6921
~/dev/ofn (master) $ find engines -name *.rb | xargs cat | wc -l
9220

Also, 54% looks really low, I wonder if this includes feature specs (a lot of our tests are feature specs)?

**Comment by luisramos0** [comment]
And most importantly, thanks for your PR @razanjoshi welcome to OFN. Join us on slack!

File: spec/coverage_helper.rb (hunk 1 above)
Comment (razanjoshi) [reply spec/coverage_helper.rb:13]: @luisramos0 Added these above-mentioned directories. Ya, this is not yet covering the features spec hence showing LOCs less than what you mentioned here. Not sure how cucumber specs are integrated here so might be worth creating another PR for including feature specs to the coverage too.

> Nice, I think this is a good thing to have.
> 
> I just wonder what are those 17klocs...
> 
> I see at least ~40klocs:
> ~/dev/ofn (master) $ find app -name *.rb | xargs cat | wc -l
> 24490
> ~/dev/ofn (master) $ find lib -name *.rb | xargs cat | wc -l
> 6921
> ~/dev/ofn (master) $ find engines -name *.rb | xargs cat | wc -l
> 9220
> 
> Also, 54% looks really low, I wonder if this includes feature specs (a lot of our tests are feature specs)?

Comment (luisramos0) [reply spec/coverage_helper.rb:13]: nice, thanks!

**Review by luisramos0 — APPROVED** [review]
I think it's even good that it doesnt include feature specs in the coverage stats. That way it will make us improve our test pyramid and get more code tested with unit tests :+1:

**Review by sauloperez — APPROVED** [review]
Thanks a lot @razanjoshi ! I didn't know Simplecov could do this!

**Comment by sauloperez** [comment]
> That way it will make us improve our test pyramid and get more code tested with unit tests

I'd comment it over that Simplecov config because it could be confusing for other devs at first. (I might as well forget in two weeks time :joy: )

**Pull request merged**